src/adapters/Auth/index.ts

```diff
@@ -0,0 +1,33 @@
+import requestManager from 'lib/requestManager';
+
+const AuthAdapter = {
+  login: (email: string, password: string) => {
+    const data = {
+      grantType: 'password',
+      email,
+      password,
+      clientId: process.env.REACT_APP_API_CLIENT_ID,
+      clientSecret: process.env.REACT_APP_API_CLIENT_SECRET,
+    };
+    return requestManager('POST', '/api/v1/oauth/token', { data });
+  },
+  refresh: (refreshToken: string) => {
+    const data = {
+      grantType: 'refresh_token',
+      refreshToken: refreshToken,
+      clientId: process.env.REACT_APP_API_CLIENT_ID,
+      clientSecret: process.env.REACT_APP_API_CLIENT_SECRET,
+    };
+    return requestManager('POST', '/api/v1/oauth/token', { data });
+  },
+  logOut: (token: string) => {
+    const data = {
+      token: token,
+      clientId: process.env.REACT_APP_API_CLIENT_ID,
+      clientSecret: process.env.REACT_APP_API_CLIENT_SECRET,
+    };
+    return requestManager('POST', '/api/v1/oauth/revoke', { data });
+  },
+};
+
+export default AuthAdapter;
```

Comment:
What is the advantage of putting all functions inside the `AuthAdapter`? Why don't you create separate functions and export them? Then in the destination file, we just need to import the needed function instead of 3 functions at the same time.
```suggestion
 const logIn = (email: string, password: string) => {
    const data = {
      grantType: 'password',
      email,
      password,
      clientId: process.env.REACT_APP_API_CLIENT_ID,
      clientSecret: process.env.REACT_APP_API_CLIENT_SECRET,
    };
    return requestManager('POST', '/api/v1/oauth/token', { data });
  };
  
  const refreshToken = (refreshToken: string) => {
    const data = {
      grantType: 'refresh_token',
      refreshToken: refreshToken,
      clientId: process.env.REACT_APP_API_CLIENT_ID,
      clientSecret: process.env.REACT_APP_API_CLIENT_SECRET,
    };
    return requestManager('POST', '/api/v1/oauth/token', { data });
  };
  
  const logOut = (token: string) => {
    const data = {
      token: token,
      clientId: process.env.REACT_APP_API_CLIENT_ID,
      clientSecret: process.env.REACT_APP_API_CLIENT_SECRET,
    };
    return requestManager('POST', '/api/v1/oauth/revoke', { data });
  };

export {
  logIn,
  refreshToken,
  logOut,
};
```

Reply:
I was confused when I created this file with this error
<img width="711" alt="image" src="https://user-images.githubusercontent.com/122268989/220806198-ec35c1b1-6931-4470-83e4-2cf5972a69ad.png">
Now I know why thanks for enlightening me 😂